src/sentry/api/authentication.py

```diff
@@ -550,3 +551,87 @@ def authenticate_token(self, request: Request, token: str) -> tuple[Any, Any]:
         sentry_sdk.get_isolation_scope().set_tag("rpc_auth", True)
 
         return (AnonymousUser(), token)
+
+
+def compare_service_signature(
+    url: str,
+    body: bytes,
+    signature: str,
+    shared_secret_setting: list[str] | None,
```

Comment:
Is there any reason to allow `None` here? It'd be safer to enforce a key explicitly being set.
We should also not allow `shared_secret_setting = [""]`. This will effectively sign with an empty HMAC key.